### PR TITLE
Add nested-limit and field exclude options

### PR DESCRIPTION
**Pull request by skoch98**
This pull request contains two improvements / adjustments.

The first commit provides two new configurations to continue using the AOS in case some classes with objectbricks ever get a little bigger.

For the second commit I barely had a little time to implement Issue #107. It is about the exclusion of fields in the search:

Without an exclude:
aos_exclude_field_with_data

With an exclude:
aos_exclude_no_field

I think a note in the documentation would be helpful from which version is this available.

### CODE REVIEW

File: src/Service.php [hunk 1]
```diff
@@ -766,4 +792,23 @@ public function extractIdsFromResult($searchResult)
...
+        $excludeFields = $this->elasticSearchConfigService->getIndexConfiguration('exclude_fields');
+
+        if (
+            isset($excludeFields[$className])
+            && in_array($fieldName, $excludeFields[$className])
+        ) {
+            return true;
+        }
+
+        return false;
```
Comment (jdreesen) [inline src/Service.php]:
```suggestion
        return isset($excludeFields[$className])
            && in_array($fieldName, $excludeFields[$className]);
```

Comment (skoch98) [reply src/Service.php]: Thanks!

File: src/Tools/ElasticSearchConfigService.php [hunk 2]
```diff
@@ -76,6 +84,18 @@ public function setIndexNamePrefix(string $indexNamePrefix): void
...
+    /**
+     * @return string|int|null
+     */
+    public function getIndexConfiguration(string $key)
+    {
+        if (isset($this->indexConfiguration[$key])) {
+            return $this->indexConfiguration[$key];
+        }
+
+        return null;
```
Comment (jdreesen) [inline src/Tools/ElasticSearchConfigService.php]:
```suggestion
        return $this->indexConfiguration[$key] ?? null;
```

**Comment by skoch98** [comment]
The ci job cant pull my forked branch. What is the correct workflow for this?

image

**Comment by fashxp** [comment]
We need to fix that on our end (restrict it for PRs) - just ignore for now

**Comment by vmalyk** [comment]
@skoch98 , @fashxp ,

I've small suggestion for extending logic and add  excluding all objects by ClassName.   I can extend PR and add  to  index_configuration new node - exclude_classes and check className in Service for next cases:
- exclude reindex objects
- remove index if class name in exclude list

I think it'll add flexible configuration and releasing excluding for fields and classes together - it'll be perfect.

Please review,

**Comment by skoch98** [comment]
> @skoch98 , @fashxp ,
> 
> I've small suggestion for extending logic and add excluding all objects by ClassName. I can extend PR and add to index_configuration new node - exclude_classes and check className in Service for next cases:
> 
> * exclude reindex objects
> * remove index if class name in exclude list
> 
> I think it'll add flexible configuration and releasing excluding for fields and classes together - it'll be perfect.
> 
> Please review,

Hi @vmalyk 

sounds good! If you need help feel free to ask me :-)

I shoud have given you rights to my forked repo ( not so familiar with github, I use gitlab ^^)

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -44,6 +44,25 @@ public function getConfigTreeBuilder()
...
+                        ->integerNode('total_fields_limit')
+                            ->defaultValue(100000)
+                        ->end()
+                        ->arrayNode('exclude_fields')
+                            ->useAttributeAsKey('class')
+                                ->prototype('scalar')
+                            ->end()
+                            ->arrayPrototype('es_hosts')
+                                ->prototype('scalar')->end()
+                            ->end()
```
Comment (skoch98) [inline src/DependencyInjection/Configuration.php]: Wrong naming. I fix it on the weekend.

Comment (skoch98) [reply src/DependencyInjection/Configuration.php]: Fixed

**Comment by fashxp** [comment]
great guys, can you just give me a ping when you are ready? then I'll review.

**Comment by vmalyk** [comment]
@fashxp, I think it's ready for review.

**Comment by skoch98** [comment]
@fashxp, I also think it's ready for review :-)

**Comment by fashxp** [comment]
looks good to me. thank you gentlemen!

**Pull request merged**